services/src/modules/plugins/logging.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  ApolloServerPlugin,
+  GraphQLRequestContext,
+  GraphQLRequestContextWillSendResponse,
+} from 'apollo-server-plugin-base';
+import { FastifyInstance } from 'fastify';
+import { Counter, Histogram } from 'prom-client';
+import logger from '../logger';
+
+let errorsInResponseCounter: Counter<string>;
+let requestDurationHistogram: Histogram<string>;
+
+export function createLoggingPlugin(fastifyInstance: FastifyInstance): ApolloServerPlugin {
+  return {
+    serverWillStart() {
+      const promClient = fastifyInstance.metrics?.client;
```

Comment:
Using `serverWillStart` to initialize metrics means that whenever theres a new ResourceGroup, the metrics will be re-initialized. I think they should be global instead, or scoped to the fastify app if that makes it easier for testing.

Reply:
The client is created in `fastify-metrics` plugin. I saw that metrics are incremented. I'll check it. Maybe existence check will be enough.